src/tools/GridTools.js

```diff
@@ -0,0 +1,70 @@
+const FILTER_SEPARATOR = ':::';
+const VALUE_SEPARATOR = ':';
+const OPERATOR_SEPARATOR = '|||';
+
+const parseGridFilters = (gridFiltersString) => {
+    const adaptGridFilters = (gridFilter = {}) => {
+        const value = gridFilter && gridFilter.value.split(OPERATOR_SEPARATOR);
+        return {
+            fieldName: gridFilter.code,
+            fieldValue: value[0],
+            joiner: 'AND',
+            operator: value[1] || '='
+        }
+    }
+
+    try {
+        return gridFiltersString ?
+            gridFiltersString.split(FILTER_SEPARATOR)
+                .map(gridFilter =>  (([code, value]) => ({code, value}))(gridFilter.split(VALUE_SEPARATOR)))
+                .map(adaptGridFilters)
+            : [];
+    } catch (err) {
+        return [];
+    }
+}
+
+const stringifyGridFilter = gridFilter => {
+    return gridFilter.fieldName + VALUE_SEPARATOR + gridFilter.fieldValue + OPERATOR_SEPARATOR + gridFilter.operator
+}
+
+const stringifyGridFilters = (gridFilters = []) => {
+    return gridFilters.map(stringifyGridFilter).join(FILTER_SEPARATOR)
+}
+
+const replaceUrlParam = (key, val) => {
+    const params = window.location.href
+            .replace(/[^?]*/, '')
+            .split(/[&?]/)
+            .filter(el => el[0])
+            .map(c => c.split("="))
+            .reduce((acc, el) => {acc[el[0]] = el[1]; return acc;}, {})
+            ;
+    params[key] = encodeURIComponent(val);
+    return '?' +
+        Object.entries(params)
+            .map(el => el[0] + '=' + el[1])
+            .reduce((acc, el) => acc ? (acc + '&' + el ) : el, '')
+        ;
+}
```

Comment:
Simplifying and using a library are not the same thing. Nonetheless, will update

Reply:
2 further comments: 

- this will make the dependency needed in a Tools file;
- your solution is not equivalent since it changes the order of the parameters.